src/components/CreateHospital.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { Mutation } from 'react-apollo';
+import gql from 'graphql-tag';
+import { ALL_HOSPITALS_QUERY } from './HospitalTable';
+
+const ADD_HOSPITAL_MUTATION = gql`
+    mutation PostMutation($name: String!) {
+        addHospital(name: $name){
+            id
+            name
+            createdAt
+            updatedAt
+        }
+    }
+`;
+
+class CreateHospital extends Component {
+    state = {
+        name: '',
+    };
+
+    render() {
+        const { name } = this.state;
+        return (
+            <div>
+                <div>
+                    <input
+                        className="mb2"
+                        value={name}
+                        onChange={e => this.setState({ name: e.target.value })}
+                        type="text"
+                        placeholder="Hospital Name"
+                    />
+                </div>
+                <Mutation
```

Comment:
I haven't looked into this way of writing mutations much yet, but I think the newer approach is to use the React hook `useMutation`. See [here](https://www.apollographql.com/docs/react/data/mutations/) for more info. I'm not sure what the pros and cons of both methods are - I'll look into that - but it seems that the React hooks might be better practice, it's definitely newer.

Reply:
i haven't used hooks too much but i'll look into it, does seem to be more widely used nowadays